raft/src/test/java/org/apache/kafka/raft/internals/BatchAccumulatorTest.java

```diff
@@ -65,6 +66,34 @@
         );
     }
 
+    @Test
+    public void testLeaderChangeMessageWritten() {
```

Comment:
A couple extra test cases to add:

1. Can we add a test case for `flush()` to ensure that it forces an immediate drain?
2. Can we add a test case in which we have undrained data when `appendLeaderChangeMessage` is called?

Reply:
@hachikuji Why would 2 ever happen in practice?